src/MongoDBBusServiceProvider.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace MongoDB\Laravel;
+
+use Illuminate\Bus\BatchFactory;
+use Illuminate\Bus\BatchRepository;
+use Illuminate\Bus\BusServiceProvider;
+use Illuminate\Container\Container;
+use Illuminate\Contracts\Support\DeferrableProvider;
+use Illuminate\Support\ServiceProvider;
+use MongoDB\Laravel\Bus\MongoBatchRepository;
+
+class MongoDBBusServiceProvider extends ServiceProvider implements DeferrableProvider
+{
+    /**
+     * Register the service provider.
+     */
+    public function register()
+    {
+        $this->app->singleton(MongoBatchRepository::class, function (Container $app) {
+            return new MongoBatchRepository(
+                $app->make(BatchFactory::class),
+                $app->make('db')->connection($app->config->get('queue.batching.database')),
```

Comment:
I realize this isn't your fault, but it's bizarre to me that the config option is `queue.batching.database` when we're actually selecting a connection. I expect this only affects the MongoDB integration, though.